### ISSUE TITLE
Bugs on the Legion Slim 5 Gen 8 (Ryzen 7 7840HS/RTX 4060 Laptop)

### ISSUE THREAD

**Issue by sentfromthehub** [BartoszCichecki/LenovoLegionToolkit#903]
### Rules

- [ ] I made myself familiar with the README.
- [ ] I checked closed issues for similar ones and did not find a similar problem mentioned before.
- [ ] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.15.4

### OS

Windows 11 Version 10.0.22621 Build 22621

### Device

Legion Slim 5 16APH8

### BIOS version

M3CN34WW

### What's wrong?

1. The dGPU doesn't actually turn off completely when the laptop is turned on from a complete shutdown in both Hybrid-Auto and Hybrid-iGPU modes. Instead, it stays in "Powered Off" or "Inactive" mode, eating up an extra 10-15 W on the discharge rate. I've tried running the kill apps and restarting dGPU functions with no success. Manage Display Mode in the NVIDIA Control Panel is set to Automatic Select. Current Status: Integrated Graphics
Attached for this issue is yesterday's log, where I, in succession, switched from Hybrid-iGPU to Hybrid-Auto and vice versa multiple times and also ran the kill apps and restart GPU functions.

<img width="548" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/121725874/f28c6d4f-a007-40c7-b800-d36f2b8e6151">

2. Toolkit eats up additional battery power. This issue is a coin toss, happening about 3 or 4 times in the last 3 weeks or so when off AC and the dGPU issue mentioned above is not happening, like today, for example. I've attached a separate log of this issue while Toolkit was running in the background without touching anything.

<img width="752" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/121725874/01ab0327-0c76-4e9c-ab7b-59b9c44119c7">

3. This is a pretty small bug, but when adding a resolution step to the "When AC Power is connected" and "When AC Power is disconnected" sections in the Actions pane after saving, the resolutions added are cleared after my Legion is shut down..

<img width="1044" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/121725874/b62a1075-b8f1-43fd-b363-b3af3a2d3ea5">

### How to reproduce the bug?

**For Issue 1:**

1. Ensure that the Manage Display Mode in the NVIDIA Control Panel is set to "Automatic Select."
2. Shut down the laptop with either Hybrid-Auto or Hybrid-iGPU mode selected
3. Turn on the laptop and sign in
4. Check the dGPU status to confirm if it remains in "Powered Off" or "Inactive" mode, rather than turning off completely.

**For Issue 2:**

1. Basically the same as Issue 1 but to confirm battery draw, I check the Battery Usage tab in Power, sleep and battery settings

**For Issue 3:**

1. Open the Actions pane in Toolkit
2. Navigate to the "When AC Power is connected" section and add a resolution step. For example, 2560 x 1600
3. Similarly, navigate to the "When AC Power is disconnected" section and add a resolution step. For example, 1920 x 1200.
4. Save the changes made to the Actions pane.
5. Confirm that the resolutions are successfully added and displayed as expected.
6. Shut down or restart the laptop.
7. Turn on the laptop again and access the Actions pane.
8. Check if the previously added resolutions are present

### What is the behavior that you expected?

For Issue 1, I expected the Discrete GPU and Overclock GPU panes to not remain visible in the Graphics section. Additionally, I expected the dGPU to turn off completely when the laptop is turned on from a complete shutdown in both Hybrid-Auto and Hybrid-iGPU modes. Additionally, I should not be able to open NVIDIA Control Panel while off AC and there should be no NVIDIA processes running in Task Manager

For Issue 2, I expected Toolkit to have a minimal impact on battery power when running in the background without any active interaction. It should also not be the top spot for battery draw, but should be near the middle or bottom of the Battery Usage section in Power, sleep and battery settings

For Issue 3, I expected the resolution steps added to the "When AC Power is connected" and "When AC Power is disconnected" sections in the Actions pane to be saved and retained even after shutting down my Legion laptop. 

### Logs

**For Issue 1:**

[log_2023_07_26_19_16_14.txt](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/12186178/log_2023_07_26_19_16_14.txt)

**For Issue 2:**

[log_2023_07_27_14_05_52.txt](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/12186179/log_2023_07_27_14_05_52.txt)

**Comment by sentfromthehub** [comment]
<img width="391" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/121725874/cb3c6661-6445-417d-8f10-a56643c905b4">


Thanks for your quick response. Focusing on the first issue, I thought the same at first too, but it seems quite annoying to pinpoint which app is using it. I turned on "Display GPU Activity in Notification Area" and nothing seems to be keeping the dGPU idling

**Comment by BartoszCichecki** [comment]
If it doesn't disconnect it is kept alive either by an app or the driver nothing I can do here.

**Comment by sentfromthehub** [comment]
<img width="558" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/121725874/2a223f03-00bc-4c08-badc-d249d21f2418">


Quick update. I've found what's keeping the dGPU idling with some Task Manager tinkering. These are the same processes that I had a problem with before and to fix it I found that turning off "Hardware-accelerated GPU scheduling" in Windows 11 "Graphics settings" is what finally fixed the issue for me. I would appreciate any thoughts even though this goes beyond Toolkit development.
